tooling/nargo_cli/src/cli/debug_cmd.rs

```diff
@@ -66,44 +75,70 @@ pub(crate) fn run(
         .expression_width
         .unwrap_or_else(|| backend.get_backend_info_or_default());
 
-    let mut workspace_file_manager = file_manager_with_stdlib(std::path::Path::new(""));
-    insert_all_files_for_workspace_into_file_manager(&workspace, &mut workspace_file_manager);
-    let mut parsed_files = parse_all(&workspace_file_manager);
-
     let Some(package) = workspace.into_iter().find(|p| p.is_binary()) else {
         println!(
             "No matching binary packages found in workspace. Only binary packages can be debugged."
         );
         return Ok(());
     };
 
-    let debug_instrumenter =
-        instrument_package_files(&mut parsed_files, &workspace_file_manager, package);
-
-    let compilation_result = compile_program_with_debug_instrumenter(
-        &workspace_file_manager,
-        &parsed_files,
+    let compiled_program = compile_bin_package_for_debugging(
+        &workspace,
         package,
-        &args.compile_options,
-        None,
-        debug_instrumenter,
-    );
-
-    let compiled_program = report_errors(
-        compilation_result,
-        &workspace_file_manager,
-        args.compile_options.deny_warnings,
-        args.compile_options.silence_warnings,
+        acir_mode,
+        skip_instrumentation,
+        args.compile_options.clone(),
     )?;
 
     let compiled_program = nargo::ops::transform_program(compiled_program, expression_width);
 
     run_async(package, compiled_program, &args.prover_name, &args.witness_name, target_dir)
 }
 
+pub(crate) fn compile_bin_package_for_debugging(
+    workspace: &Workspace,
+    package: &Package,
+    acir_mode: bool,
+    skip_instrumentation: bool,
+    compile_options: CompileOptions,
+) -> Result<CompiledProgram, CompileError> {
+    let mut workspace_file_manager = file_manager_with_stdlib(std::path::Path::new(""));
+    insert_all_files_for_workspace_into_file_manager(workspace, &mut workspace_file_manager);
+    let mut parsed_files = parse_all(&workspace_file_manager);
+
+    let compile_options = CompileOptions {
+        instrument_debug: !skip_instrumentation,
+        force_brillig: !acir_mode,
+        ..compile_options
+    };
+
+    let compilation_result = if !skip_instrumentation {
+        let debug_state =
+            instrument_package_files(&mut parsed_files, &workspace_file_manager, package);
+
+        compile_program_with_debug_instrumenter(
```

Comment:
Nit: can we add instrumenter to the cspell so we don't see the github action warnings.

Reply:
Done!